### ISSUE TITLE
AssetsImpl errors on Windows

### ISSUE THREAD

**Issue by SKBotNL** [haxeui/haxeui-openfl#62]
## Expected Behavior
The resource should load

## Current Behavior
When using an external image/web address as the resource of an image the built program for Windows errors with:\
`haxe/ui/backend/AssetsImpl.hx:95: [IOErrorEvent type="ioError" bubbles=true cancelable=false text="Unknown error" errorID=0]`

## Steps to Reproduce (for bugs)

1. Create an Image
2. Build the project for Windows
3. See error

## Test app / minimal test case
```haxe
package;

import haxe.ui.containers.VBox;
import haxe.ui.components.Image;
import haxe.ui.HaxeUIApp;

class Main {
    public static function main() {
        var app = new HaxeUIApp();
        app.ready(function() {
            var vbox = new VBox();
            vbox.percentHeight = 100;
            vbox.percentWidth = 100;

            var image = new Image();
            image.resource = "https://cdn.discordapp.com/icons/613425648685547541/a_08b73e86d3c5661db3e397e4c557f751.webp";
            image.width = 128;
            image.height = 128;

            vbox.addComponent(image);
            app.addComponent(vbox);
            app.start();
        });
    }
}


```

## Your Environment

* Version used: 1.6.0
* Operating System and version (desktop or mobile): Windows 10

**Comment by ianharrigan** [comment]
OK, so can repro with the webp, which i think is probably valid because im guessing it doesnt support webp... although the error seems a little odd... (ill look into it).

I can repro with a png, i tried: `https://upload.wikimedia.org/wikipedia/commons/7/70/Example.png` and it seemed fine... what url were you using?

Cheers,
Ian

**Comment by SKBotNL** [comment]
> https://upload.wikimedia.org/wikipedia/commons/7/70/Example.png

That png indeed works, others such as `https://via.placeholder.com/600x400.png` and `https://filesamples.com/samples/image/png/sample_1280%C3%97853.png` don't work

**Comment by ianharrigan** [comment]
![image](https://user-images.githubusercontent.com/3024242/234360743-4b493c3e-0d44-4e81-975d-72f0242f5e13.png)

Seems fine this end... ill update all my libs, something must have changed? What versions of openfl, lime, haxe, etc are you running?

**Comment by SKBotNL** [comment]
> ![image](https://user-images.githubusercontent.com/3024242/234360743-4b493c3e-0d44-4e81-975d-72f0242f5e13.png)
> 
> Seems fine this end... ill update all my libs, something must have changed? What versions of openfl, lime, haxe, etc are you running?

Can you share your code?
Versions:
- Haxe: 4.3.0
- OpenFL: 9.2.1
- Lime: 8.0.1

**Comment by SKBotNL** [comment]
Actually, after some more testing, it seems that in this cast html5 is actually having an issue

**Comment by ianharrigan** [comment]
right, that also makes sense i think the browser is preventing haxe (not haxeui) from making that http request becuase the requested resources has said (CORS):

```
Access to XMLHttpRequest at 'https://filesamples.com/samples/image/png/sample_1280%C3%97853.png' from origin 'http://127.0.0.1:3000' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
``` 

```
Access to XMLHttpRequest at 'https://via.placeholder.com/600x400.png' from origin 'http://127.0.0.1:3000' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
```

It make work if you deployed to a real domain rather than localhost

**Comment by SKBotNL** [comment]
> right, that also makes sense i think the browser is preventing haxe (not haxeui) from making that http request becuase the requested resources has said (CORS):
> 
> ```
> Access to XMLHttpRequest at 'https://filesamples.com/samples/image/png/sample_1280%C3%97853.png' from origin 'http://127.0.0.1:3000' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
> ```
> 
> ```
> Access to XMLHttpRequest at 'https://via.placeholder.com/600x400.png' from origin 'http://127.0.0.1:3000' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
> ```
> 
> It may work if you deployed to a real domain rather than localhost (but probably not)

That makes sense, is there any way to add a `Access-Control-Allow-Origin` header to the request?

**Comment by ianharrigan** [comment]
well, `Access-Control-Allow-Origin` comes from the server, not the browser, so the short answer is nope, it aint gonna happen. Maybe the http server you _actually_ want to get images for is under your control? In which case you should be fine.

**Comment by SKBotNL** [comment]
> well, `Access-Control-Allow-Origin` comes from the server, not the browser, so the short answer is nope, it aint gonna happen. Maybe the http server you _actually_ want to get images for is under your control? In which case you should be fine.

You're right. Lucky that is indeed the case.
